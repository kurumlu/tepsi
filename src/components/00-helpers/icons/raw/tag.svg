<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M2 11.8484V3.875C2 2.83945 2.83945 2 3.875 2H11.8484C12.3456 2 12.8226 2.19755 13.1742 2.54918L21.4508 10.8258C22.1831 11.5581 22.1831 12.7452 21.4508 13.4775L13.4775 21.4508C12.7452 22.1831 11.5581 22.1831 10.8258 21.4508L2.54918 13.1742C2.19755 12.8226 2 12.3456 2 11.8484H2ZM6.375 4.5C5.33945 4.5 4.5 5.33945 4.5 6.375C4.5 7.41055 5.33945 8.25 6.375 8.25C7.41055 8.25 8.25 7.41055 8.25 6.375C8.25 5.33945 7.41055 4.5 6.375 4.5Z" fill="black"/>
</svg>
