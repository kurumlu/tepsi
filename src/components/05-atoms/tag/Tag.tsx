import * as React from 'react';
import { FunctionComponent } from 'react';
import cn from 'classnames';

export enum TagColor {
  PRIMARY = 'primary', // 1st
  SECONDARY = 'secondary', // 2nd
  TERTIARY = 'tertiary', // 3rd
  QUATERNARY = 'quaternary', // 4th
  QUINARY = 'quinary', // 5th
  SENARY = 'senary', // 6th
  SEPTENARY = 'septenary', // 7th
  GENERAL = 'general',
  DISCOUNT = 'discount',
  SUSTAINABLE = 'sustainable',
}

export type TagProps = {
  color?: TagColor;
  dataQA?: string;
  className?: string;
};

export const Tag: FunctionComponent<TagProps> = ({
  children,
  color,
  dataQA,
  className,
}) => (
  <div
    className={cn('sd-a-tag', { [`sd-a-tag--${color}`]: color }, className)}
    data-qa={dataQA}>
    {children}
  </div>
);
