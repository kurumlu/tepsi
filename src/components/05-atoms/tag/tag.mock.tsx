import * as React from 'react';
import { FunctionComponent } from 'react';
import { Icon, IconName, IconSize } from '../icon/Icon';

export const MockChildren: FunctionComponent = () => (
  <Icon name={IconName.FEMALE} size={IconSize.MEDIUM} />
);
