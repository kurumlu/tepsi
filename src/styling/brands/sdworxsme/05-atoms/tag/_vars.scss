$a-tag--background: $t2-color__white;
$a-tag--color: $t2-color__gray-100;
