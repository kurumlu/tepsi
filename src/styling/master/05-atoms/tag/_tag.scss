// Variables for Tag ATOM
$a-tag--height: get-spacing(3) !default;
$a-tag--padding: get-spacing(.25) get-spacing(1) !default;

$a-tag--font-map: $t2-font-map__body !default;
$a-tag--font-size: $t2-font-size__body-s !default;
$a-tag--line-height: $t2-line-height__s !default;

$a-tag--background: get-brand-color() !default;
$a-tag--color: $t2-color__white !default;

// --primary
$a-tag--primary--background: get-brand-color(primary) !default;
$a-tag--primary--color: $t2-color__white !default;

// --secondary
$a-tag--secondary--background: get-brand-color(secondary) !default;
$a-tag--secondary--color: $t2-color__white !default;

// --tertiary
$a-tag--tertiary--background: get-brand-color(tertiary) !default;
$a-tag--tertiary--color: $t2-color__white !default;

// --quaternary
$a-tag--quaternary--background: get-brand-color(quaternary) !default;
$a-tag--quaternary--color: $t2-color__white !default;

// --quinary
$a-tag--quinary--background: get-brand-color(quinary) !default;
$a-tag--quinary--color: $t2-color__white !default;

// --senary
$a-tag--senary--background: get-brand-color(senary) !default;
$a-tag--senary--color: $t2-color__white !default;

// --septenary
$a-tag--septenary--background: get-brand-color(septenary) !default;
$a-tag--septenary--color: $t2-color__white !default;

// --general (general: gender, -x%, ...)
$a-tag--general--background: $t2-color__white !default;
$a-tag--general--color: $t2-color__black !default;

// --discount (discount)
$a-tag--discount--background: $t2-color__discount !default;
$a-tag--discount--color: $t2-color__white !default;

// --sustainable
$a-tag--sustainable--background: $t2-color__sustainable !default;
$a-tag--sustainable--color: $t2-color__white !default;


// Code for Tag ATOM
.as-a-tag {
  @include font-map($a-tag--font-map);
  box-sizing: border-box;
  display: inline-block;
  height: $a-tag--height;
  padding: $a-tag--padding;
  color: $a-tag--color;
  font-size: $a-tag--font-size;
  line-height: $a-tag--line-height;
  background: $a-tag--background;
}

// --tag-color
.as-a-tag--primary {
  color: $a-tag--primary--color;
  background: $a-tag--primary--background;
}

.as-a-tag--secondary {
  color: $a-tag--secondary--color;
  background: $a-tag--secondary--background;
}

.as-a-tag--tertiary {
  color: $a-tag--tertiary--color;
  background: $a-tag--tertiary--background;
}

.as-a-tag--quaternary {
  color: $a-tag--quaternary--color;
  background: $a-tag--quaternary--background;
}

.as-a-tag--quinary {
  color: $a-tag--quinary--color;
  background: $a-tag--quinary--background;
}

.as-a-tag--senary {
  color: $a-tag--senary--color;
  background: $a-tag--senary--background;
}

.as-a-tag--septenary {
  color: $a-tag--septenary--color;
  background: $a-tag--septenary--background;
}

.as-a-tag--general {
  color: $a-tag--general--color;
  background: $a-tag--general--background;
}

.as-a-tag--discount {
  color: $a-tag--discount--color;
  background: $a-tag--discount--background;
}

.as-a-tag--sustainable {
  color: $a-tag--sustainable--color;
  background: $a-tag--sustainable--background;
}
