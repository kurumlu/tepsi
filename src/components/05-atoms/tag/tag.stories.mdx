import { Meta, Story, Canvas } from '@storybook/addon-docs/blocks';
import { Tag, TagColor } from './Tag';
import { Icon, IconName } from '../icon/Icon';

<Meta
  title="Design system/Patterns/Atoms/Tag"
  component={Tag}
/>

# Tag

> Tags show the user information about something.
> They are typically placed next or on top of another element, and attract attention.

<Canvas>
  <Story name="Default">
    <Tag>
      Default
    </Tag>
  </Story>
</Canvas>

---

## Available props

| Name        | Type       | Required | Default value |
|-------------|------------|----------|---------------|
| `color`     | *TagColor* |          |               |
| `dataQA`    | *string*   |          |               |
| `className` | *string*   |          |               |

---

### Color

Use the `color` prop to set the color of the tag. Uses the `TagColor` enum.

| TagColor      |
|---------------|
| `PRIMARY`     |
| `SECONDARY`   |
| `TERTIARY`    |
| `QUATERNARY`  |
| `QUINARY`     |
| `SENARY`      |
| `SEPTENARY`   |
| `GENERAL`     |
| `DISCOUNT`    |
| `SUSTAINABLE` |

```javascript
<Tag color={...}>...</Tag>
```

#### Default (Online-exclusive)

```javascript
<Tag>...</Tag>
```

<Canvas>
  <Story name="Color - default">
    <Tag>
      Default
    </Tag>
  </Story>
</Canvas>

#### Primary

```javascript
<Tag color={TagColor.PRIMARY}>...</Tag>
```

<Canvas>
  <Story name="Color - primary">
    <Tag color={TagColor.PRIMARY}>
      Primary
    </Tag>
  </Story>
</Canvas>

#### Secondary

```javascript
<Tag color={TagColor.SECONDARY}>...</Tag>
```

<Canvas>
  <Story name="Color - secondary">
    <Tag color={TagColor.SECONDARY}>
      Secondary
    </Tag>
  </Story>
</Canvas>

#### Tertiary

```javascript
<Tag color={TagColor.TERTIARY}>...</Tag>
```

<Canvas>
  <Story name="Color - tertiary">
    <Tag color={TagColor.TERTIARY}>
      Tertiary
    </Tag>
  </Story>
</Canvas>

#### Quaternary

```javascript
<Tag color={TagColor.QUATERNARY}>...</Tag>
```

<Canvas>
  <Story name="Color - quaternary">
    <Tag color={TagColor.QUATERNARY}>
      Quaternary
    </Tag>
  </Story>
</Canvas>

#### Quinary

```javascript
<Tag color={TagColor.QUINARY}>...</Tag>
```

<Canvas>
  <Story name="Color - quinary">
    <Tag color={TagColor.QUINARY}>
      Quinary
    </Tag>
  </Story>
</Canvas>

#### SENARY

```javascript
<Tag color={TagColor.SENARY}>...</Tag>
```

<Canvas>
  <Story name="Color - senary">
    <Tag color={TagColor.SENARY}>
      Senary
    </Tag>
  </Story>
</Canvas>

#### SEPTENARY

```javascript
<Tag color={TagColor.SEPTENARY}>...</Tag>
```

<Canvas>
  <Story name="Color - septenary">
    <Tag color={TagColor.SEPTENARY}>
      Septenary
    </Tag>
  </Story>
</Canvas>

#### GENERAL

```javascript
<Tag color={TagColor.GENERAL}>...</Tag>
```

<Canvas>
  <Story name="Color - general">
    <Tag color={TagColor.GENERAL}>
      General
    </Tag>
  </Story>
</Canvas>

#### DISCOUNT

```javascript
<Tag color={TagColor.DISCOUNT}>...</Tag>
```

<Canvas>
  <Story name="Color - discount">
    <Tag color={TagColor.DISCOUNT}>
      Discount
    </Tag>
  </Story>
</Canvas>

#### Sustainable

```javascript
<Tag color={TagColor.SUSTAINABLE}>...</Tag>
```

<Canvas>
  <Story name="Color - sustainable">
    <Tag color={TagColor.SUSTAINABLE}>
      Sustainable
    </Tag>
  </Story>
</Canvas>

---

### Other

Other props that can be added:
- `dataQA`
