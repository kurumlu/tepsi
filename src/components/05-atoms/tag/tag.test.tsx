import React from 'react';
import { mount, shallow, ShallowWrapper } from 'enzyme';

import { Tag, TagColor, TagProps } from './Tag';
import { MockChildren } from './tag.mock';

describe('renders <Tag />', () => {
  const component: ShallowWrapper<TagProps> = shallow(<Tag>-30%</Tag>);

  it('renders <Tag /> component', () => {
    expect(component).toMatchSnapshot('default');
  });

  it('has specific class depends on color', () => {
    const TagWithSecondColor: ShallowWrapper<TagProps> = shallow(
      <Tag color={TagColor.SECONDARY}>-30%</Tag>
    );
    expect(TagWithSecondColor).toMatchSnapshot('color');
    expect(
      TagWithSecondColor.hasClass(`sd-a-tag--${TagColor.SECONDARY}`)
    ).toBeTruthy();
  });

  it(`has data-qa attr`, () => {
    const TagWithDataAttr: ShallowWrapper<TagProps> = shallow(
      <Tag dataQA={'123'}>-30%</Tag>
    );
    expect(TagWithDataAttr).toMatchSnapshot('data-qa');
    expect(TagWithDataAttr.prop('data-qa')).toEqual('123');
  });

  it(`renders <Tag /> component with children`, () => {
    const componentWithChildren = mount(
      <Tag>
        <MockChildren />
      </Tag>
    );

    expect(componentWithChildren).toMatchSnapshot('children');
    expect(componentWithChildren.find('svg')).toHaveLength(1);
  });
});
