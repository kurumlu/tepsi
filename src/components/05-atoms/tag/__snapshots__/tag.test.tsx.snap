// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Tag /> has data-qa attr: data-qa 1`] = `
<div
  className="as-a-tag"
  data-qa="123"
>
  -30%
</div>
`;

exports[`renders <Tag /> has specific class depends on color: color 1`] = `
<div
  className="as-a-tag as-a-tag--secondary"
>
  -30%
</div>
`;

exports[`renders <Tag /> renders <Tag /> component with children: children 1`] = `
<Tag>
  <div
    className="as-a-tag"
  >
    <MockChildren>
      <Icon
        name="female"
        size="medium"
      >
        <span
          className="as-a-icon as-a-icon--female as-a-icon--medium"
          role="img"
        >
          <svg
            aria-hidden="true"
            className="as-a-icon__svg"
            focusable="false"
          />
        </span>
      </Icon>
    </MockChildren>
  </div>
</Tag>
`;

exports[`renders <Tag /> renders <Tag /> component: default 1`] = `
<div
  className="as-a-tag"
>
  -30%
</div>
`;
